src/IO/H5/CheckH5PropertiesMatch.cpp

```diff
@@ -0,0 +1,82 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "IO/H5/CheckH5PropertiesMatch.hpp"
+
+#include <cstddef>
+#include <iterator>
+
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/SourceArchive.hpp"
+#include "IO/H5/VolumeData.hpp"
+
+namespace h5 {
+// Compares source archives of each file against the 0th volume file. Returns
```

Comment:
Up to you, but we don't require the extra description comment in the `cpp` file since it should be documented in the hpp file. But if you want to keep, that's totally fine too :)